### PR TITLE
HARVESTER: PCI device UI changes

### PR DESCRIPTION
**Pull request by DaiYuzeng**
#### PR Checklist
- ~Is this a multi-tenancy feature/bug?~
    - ~[ ] Yes, the relevant RBAC changes are at:~
- ~Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?~
    - ~[ ] Yes, the relevant PR is at:~
- ~Are backend engineers aware of UI changes?~
    - ~[ ] Yes, the backend owner is:~

### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/3146

### Areas or cases that should be tested

1. Go to VM create page
2. Switch to PCI devices tab
3. Select device from dropdown
4. You can only choose devices are belonged the same node
5. You can not select the device owned by other VMs
6. Matrix should display the PCI claim name of devices

### CODE REVIEW

**Review by n313893254 — COMMENTED** [review]
@DaiYuzeng  Does the PCI dropdown need to filter by node selector?

image

image

**Comment by DaiYuzeng** [comment]
Good question, I need to confirm that. Thanks! @n313893254

File: pkg/harvester/l10n/en-us.yaml [hunk 1]
```diff
@@ -206,6 +207,8 @@ harvester:
       suffix: to enable it to manage your PCI devices.
     noPCIPermission: Please contact system admin to enable the PCI devices first.
     enablePassthroughWarning: Please be careful not to use host-owned PCI devices (e.g., management and VLAN NICs). Incorrect device allocation may cause damage to your cluster, including node failure.
+    matrixHostName: Hostname
```
Comment (n313893254) [inline pkg/harvester/l10n/en-us.yaml:210]: Should be `Host Name` ?

File: pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/CompatibilityMatrix.vue [hunk 2]
```diff
@@ -52,40 +52,28 @@ export default {
...
-
-      return allNodesWithDevice.includes(nodeName);
+    nodeHasDevice(nodeName, deviceCRD) {
+      return deviceCRD.status.nodeName === nodeName;
     },
 
-    vmsUsingDevice(id) {
-      const resourceName = this.uniqueDevices[id].deviceCRDs[0].status?.resourceName;
+    vmsUsingDevice(deviceCRD) {
+      const name = deviceCRD.status?.metadata?.name;
```
Comment (n313893254) [inline pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/CompatibilityMatrix.vue:64]:
```suggestion
      const name = deviceCRD.metadata?.name;
```

File: pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/CompatibilityMatrix.vue [hunk 3]
```diff
@@ -52,40 +52,28 @@ export default {
...
-      return this.devicesInUse[resourceName]?.usedBy || null;
+      return this.devicesInUse[name]?.usedBy || null;
     },
 
-    noneAvailable(id) {
-      const resourceName = this.uniqueDevices[id].deviceCRDs[0].status?.resourceName;
-
-      const count = this.devicesInUse[resourceName]?.count || 0;
+    noneAvailable(deviceCRD) {
+      const name = deviceCRD.status?.metadata?.name;
```
Comment (n313893254) [inline pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/CompatibilityMatrix.vue:70]:
```suggestion
      const name = deviceCRD.metadata?.name;
```

File: pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/CompatibilityMatrix.vue [hunk 4]
```diff
@@ -52,40 +52,28 @@ export default {
...
-      const allNodesWithDevice = this.uniqueDevices[deviceId]?.nodes;
-
-      return allNodesWithDevice.includes(nodeName);
+    nodeHasDevice(nodeName, deviceCRD) {
+      return deviceCRD.status.nodeName === nodeName;
     },
 
-    vmsUsingDevice(id) {
-      const resourceName = this.uniqueDevices[id].deviceCRDs[0].status?.resourceName;
+    vmsUsingDevice(deviceCRD) {
```
Comment (n313893254) [inline pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/CompatibilityMatrix.vue:63]: Seems it is an unused method. Could we remove it?

**Review by n313893254 — CHANGES_REQUESTED** [review]
1. PCI devices selector should be disabled on the detail config page
image

**Review by n313893254 — APPROVED** [review]
LGTM

**Pull request merged**